smack-experimental/src/main/java/org/jivesoftware/smackx/mam/element/MamPrefsIQ.java

```diff
@@ -0,0 +1,159 @@
+ * @see <a href="http://xmpp.org/extensions/xep-0313.html">XEP-0313: Message
+ *      Archive Management</a>
+ *
+ */
+public class MamPrefsIQ extends IQ {
+
+    /**
+     * the preferences element.
+     */
+    public static final String ELEMENT = "prefs";
+
+    /**
+     * the IQ NAMESPACE.
+     */
+    public static final String NAMESPACE = MamElements.NAMESPACE;
+
+    /**
+     * true if it is a request for update preferences.
+     */
+    private boolean isUpdate;
+
+    /**
+     * true if it is a result preferences.
+     */
+    private boolean isResult;
+
+    /**
+     * list of always.
+     */
+    private List<Jid> alwaysJids;
+
+    /**
+     * list of never.
+     */
+    private List<Jid> neverJids;
+
+    /**
+     * default field.
+     */
+    private String defaultField;
+
+    /**
+     * MAM preferences IQ constructor.
+     * 
+     * @param type
+     * @param alwaysJids
+     * @param neverJids
+     * @param defaultField
+     */
+    public MamPrefsIQ(Type type, List<Jid> alwaysJids, List<Jid> neverJids, String defaultField) {
+        super(ELEMENT, NAMESPACE);
+        this.setType(type);
+        this.isUpdate = this.getType().equals(Type.set);
+        this.isResult = this.getType().equals(Type.result);
+        this.alwaysJids = alwaysJids;
+        this.neverJids = neverJids;
+        this.defaultField = defaultField;
+    }
+
+    /**
+     * True if it is a request for update preferences.
+     * 
+     * @return the update preferences boolean
+     */
+    public boolean isUpdate() {
+        return isUpdate;
+    }
+
+    /**
+     * True if it is a result.
+     * 
+     * @return the result preferences boolean
+     */
+    public boolean isResult() {
+        return isUpdate;
+    }
+
+    /**
+     * Get the list of always store info JIDs.
+     * 
+     * @return the always list
+     */
+    public List<Jid> getAlwaysJids() {
+        return alwaysJids;
+    }
+
+    /**
+     * Get the list of never store info JIDs.
+     * 
+     * @return the never list
+     */
+    public List<Jid> getNeverJids() {
+        return neverJids;
+    }
+
+    /**
+     * Get the default field.
+     * 
+     * @return the default field
+     */
+    public String getDefault() {
+        return defaultField;
+    }
+
+    @Override
+    protected IQChildElementXmlStringBuilder getIQChildElementBuilder(IQChildElementXmlStringBuilder xml) {
+
+        if (isUpdate || isResult) {
+            xml.attribute("default", defaultField);
+        }
+
+        xml.rightAngleBracket();
+
+        if (alwaysJids != null) {
+            MamElements.AlwaysJidListElement alwaysElement = new AlwaysJidListElement(alwaysJids);
+            xml.element(alwaysElement);
+        }
+
+        if (neverJids != null) {
+            MamElements.NeverJidListElement neverElement = new NeverJidListElement(neverJids);
+            xml.element(neverElement);
+        }
+
+        return xml;
+    }
+
+    public static MamPrefsIQ from(IQ iq) {
```

Comment:
Please remove that method. We use the `from()` pattern in Smack only to retrieve extension elements. In order to downcast an IQ, we prefer explicit casts.